### PR TITLE
Adding server sync feature to the module to support import from SolusIO

### PR DESCRIPTION
**Pull request by redikultsevsilver**
I am adding the server sync feature (https://developers.whmcs.com/provisioning-modules/server-sync/) to our module. With this feature it will be possible to import virtual servers and clients from SolusIO to WHMCS.

### CODE REVIEW

File: lib/module.php [hunk 1]
```diff
@@ -629,3 +635,90 @@ function solusiovps_restart(array $params)
...
+
+        $userResource = new UserResource(Connector::create($serverParams));
+        $solusUser = $userResource->getUserByEmail($params['clientsdetails']['email']);
+
+        $serverResource = new ServerResource(Connector::create($serverParams));
+        $allServersOfUser = $serverResource->getAllByUser($solusUser['id']);
+
+        foreach ($allServersOfUser as $server) {
+            if ($server['name'] == $params['domain']) {
+                $serverPayload = $serverResource->get($server['id']);
```
Comment (kryptamine) [inline lib/module.php:704]: Can't we take serverPayload directly from the $server, if I correctly understand they have the same contract.

File: lib/module.php [hunk 2]
```diff
@@ -629,3 +635,90 @@ function solusiovps_restart(array $params)
...
+            return [
+                'success' => "Account is already synced",
+            ];
+        }
+
+        $serverId = (int)$params['serverid'];
+        $serverParams = Server::getParams($serverId);
+
+        $userResource = new UserResource(Connector::create($serverParams));
+        $solusUser = $userResource->getUserByEmail($params['clientsdetails']['email']);
```
Comment (kryptamine) [inline lib/module.php:697]: Handle the case when there is no user found by email please

File: lib/module.php [hunk 3]
```diff
@@ -629,3 +635,90 @@ function solusiovps_restart(array $params)
...
+function solusiovps_syncAccount(array $params)
+{
+    try {
+        if (!empty(SolusServer::getByServiceId($params['serviceid']))) {
+            return [
+                'success' => "Account is already synced",
+            ];
+        }
+
+        $serverId = (int)$params['serverid'];
```
Comment (kryptamine) [inline lib/module.php:693]: You can just put this variable in place, it used just once, no reason to extract it to an additional variable

`$serverParams = Server::getParams((int)$params['serverid']);`

File: lib/module.php [hunk 4]
```diff
@@ -629,3 +635,90 @@ function solusiovps_restart(array $params)
...
+{
+    try {
+        $statusMap = [
+            "0" => WHMCS\Service\Status::ACTIVE,
+            "1" => WHMCS\Service\Status::SUSPENDED,
+        ];
+        $accounts = [];
+
+        $serverId = (int)$params['serverid'];
+        $serverParams = Server::getParams($serverId);
```
Comment (kryptamine) [inline lib/module.php:649]: You can just put this variable in place, it used just once, no reason to extract it to an additional variable

`$serverParams = Server::getParams((int)$params['serverid']);`

File: lib/SolusAPI/Resources/ServerResource.php [hunk 5]
```diff
@@ -30,6 +30,11 @@ public function get(int $id): array
         return $this->processResponse($this->connector->get("servers/{$id}"));
     }
 
+    public function listServerPage(int $page): array
+    {
+        return $this->processResponse($this->connector->get("servers?page=${page}"));
```
Comment (kryptamine) [inline lib/SolusAPI/Resources/ServerResource.php:35]: `return $this->processResponse($this->connector->get("servers?page=$page"));`

File: lib/module.php (hunk 1 above)
Comment (redikultsevsilver) [reply lib/module.php]: Thanks a lot for the input! I have adjusted the code according to your comments.

**Review by kryptamine — APPROVED** [review]

**Pull request merged**